backend/inspirehep/records/api/data.py

```diff
@@ -17,3 +17,23 @@ class DataRecord(CitationMixin, InspireRecord):
     es_serializer = DataElasticSearchSchema
     pid_type = "dat"
     pidstore_handler = PidStoreData
+
+    @classmethod
+    def create(
+        cls,
+        data,
+        *args,
+        **kwargs,
+    ):
+        record = super().create(data, **kwargs)
+        record.update_refs_in_citation_table()
+        return record
+
+    def update(
+        self,
+        data,
+        *args,
+        **kwargs,
+    ):
+        self.update_refs_in_citation_table()
```

Comment:
This should be called after `super().update()` IMHO otherwise it will update data for current record not for updated one.

Reply:
True thanks!